src/http/client/response.cr

```diff
@@ -31,6 +31,7 @@ class HTTP::Client::Response
     @body
   end
 
+  # Returns true if the client's request was successfully received, understood, and accepted.
```

Comment:
I'd prefer to be explicit here and document that `success?` means that the response status code is in the range `200..299`.